### PR TITLE
Apigateway stages

### LINKED ISSUE #628: Using Simple API Gateway to deploy multiple stages
I am trying to deploy new environment using zappa deploy. 

I am getting an error like 

`botocore.exceptions.ClientError: An error occurred (BadRequestException) when calling the CreateRestApi operation: Maximum number of APIs has been reached.  Please contact AWS if you need additional APIs.`

When I raised a limit increase request in aws console

I got the reply like 

    1. If two APIs serve a similar business purpose, then merge the resources into one API.
    2. Use stages instead of creating separate dev/beta APIs.

are we going to get the support for deploying the multiple stages to same API Gateway API

### PR DESCRIPTION
**Pull request by anush0247**
fixed #628 #254

### CODE REVIEW

**Comment by anush0247** [comment]
@Miserlou any updated about this pr

**Comment by anush0247** [comment]
I am clueless how to increase the coverage so that all checks going to pass

**Comment by anush0247** [comment]
Hey @Miserlou 

We would love to have this feature in Zappa, We are waiting for this PR to be accepted. It's been nearly 17 days that submitted this PR.  Are you looking for any additional features in this PR, please do let us know? will try to implement those, that are related to supporting multiple API gateway stages deployment.

**Comment by anush0247** [comment]
@Miserlou I will fix the conflicts and push the changes in a couple of days. will this be suffice, or are you looking for any other changes as well?

File: tests/tests_placebo.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def test_upload_remove_s3(self, session):
         z = Zappa(session)
         zip_path = z.create_lambda_zip(minify=False)
         res = z.upload_to_s3(zip_path, bucket_name)
-        os.remove(zip_path)
+        # os.remove(zip_path)
```
Comment (anush0247) [inline tests/tests_placebo.py]: we can clear, I will update

**Comment by copynpaste** [comment]
How are we on this, are there any issues/is this already reviewed?

**Comment by deoresheetal** [comment]
@anush0247 are your fixes pushed?

**Comment by deoresheetal** [comment]
@anush0247 Are your fixes pushed?

**Comment by mcrowson** [comment]
Was looking through the changes and just wonder why this is an optional config. Why wouldn’t all deployments want to follow this stage pattern?

**Comment by anush0247** [comment]
@mcrowson @deoresheetal I will update the pull request and resolve the conflict I can even make this as a required configuration but, we need to think about the backward compatibility

File: zappa/core.py [hunk 2]
```diff
@@ -813,29 +842,51 @@ def update_lambda_configuration(    self,
...
+            FunctionName=function_name
         )
+        try:
+            del previous_config_response["Environment"]["Error"]
+        except KeyError as err:
+            pass
+
+        updated_dict = {
+            "FunctionName": function_name,
+            "Runtime": 'python2.7',
```
Comment (zaraken) [inline zappa/core.py]: Why is this hardcoded to python 2.7? What if I'm using 3?

**Pull request closed without merging**